gurmukhiutils/unicode.py

```diff
@@ -0,0 +1,448 @@
+from typing import Literal, Match
+
+UNICODE_STANDARDS = Literal["Unicode Consortium", "Sant Lipi"]
+
+
+def unicode(
+    string: str,
+    unicode_standard: Literal[UNICODE_STANDARDS] = "Unicode Consortium",
+) -> str:
+    """
+    Converts any ascii gurmukhi characters and sanitizes to unicode gurmukhi.
+
+    Args:
+        string: The string to affect.
+
+        unicode_standard: The mapping system to use. The default is unicode compliant and can render 99% of the Shabad OS Database. The other option "Sant Lipi" is intended for a custom unicode font bearing the same name (see: https://github.com/shabados/SantLipi). Defaults to "Unicode Consortium".
+
+    Returns:
+        A string whose Gurmukhi is normalized to a Unicode standard.
+
+    Examples:
+        >>> unicode("123")
+        '੧੨੩'
+
+        >>> unicode("<> > <")
+        'ੴ ☬ ੴ'
+
+        >>> unicode("gurU")
+        'ਗੁਰੂ'
+    """
+
+    import re
```

Comment:
imports within functions like this are generally discouraged without good reason
https://peps.python.org/pep-0008/#imports